vm/error_test.go

```diff
@@ -148,15 +155,159 @@ func TestArgumentError(t *testing.T) {
 		foo(*arr)
 		`,
 			"ArgumentError: Expect at most 3 args for method 'foo'. got: 4",
-			6, 1},
+			6, 1, 1},
+		{`def foo(a, b, c)
+		  a + b + c
+		end
+
+		def bar
+		  arr = [1, 2, 3, 5]
+		  foo(*arr)
+		end
+
+		bar
+		`,
+			"ArgumentError: Expect at most 3 args for method 'foo'. got: 4",
+			// The two objects on the stack would be:
+			// - the receiver of bar, because that call haven't been finished
+			// - the error object
+			6, 2, 2},
+		{`def foo(a, b, c)
+		  a + b + c
+		end
+
+		def bar
+		  arr = [1, 2, 3, 5]
+		  foo(*arr)
+		end
+
+		def baz
+		  bar
+		end
+
+		baz
+		`,
+			"ArgumentError: Expect at most 3 args for method 'foo'. got: 4",
+			// The three objects on the stack would be:
+			// - the receiver of baz, because that call haven't been finished
+			// - the receiver of bar, because that call haven't been finished
+			// - the error object
+			6, 3, 3},
 	}
 
 	for i, tt := range tests {
 		v := initTestVM()
 		evaluated := v.testEval(t, tt.input, getFilename())
-		checkError(t, i, evaluated, tt.expected, getFilename(), tt.errorLine)
+		checkErrorMsg(t, i, evaluated, tt.expected)
 		v.checkCFP(t, i, tt.expectedCFP)
-		v.checkSP(t, i, 1)
+		v.checkSP(t, i, tt.expectedSP)
+	}
+}
+
+func TestStackTraces(t *testing.T) {
```

Comment:
@goby-lang/maintainers We should have error's stack traces in the future! There's still some places need to be improved, but at least the basic functionality is done.